website/app/components/doc/link-with-icon/index.hbs

```diff
@@ -9,7 +9,7 @@
     @route={{@route}}
     @models={{this.models}}
     ...attributes
-    {{on "click" (fn this.eventTracking.trackEvent @eventName)}}
+    {{doc-track-event-on "click" @eventName}}
```

Comment:
general consideration, regarding the API of the modifier:
- since 99% of the cases will be on `click` what if we make this the default type of event trigger?
- in that case, it means we shouldn't use positional arguments, but named arguments

Reply:
also, but this is a personal preference, I would keep the modifier name a bit simpler (`doc-track-event`)

Reply:
That's what I started with, but I added the `on` when I changed it to take different event triggers. I was meh about it though and don't have a strong opinion. I'll change it back.